### ISSUE TITLE
Cypress commands does not execute on failed tests using `after` or `afterEach` hooks

### ISSUE THREAD

**Issue by adokania** [cypress-io/cypress#203]
I have an afterEach() block that tries to run a cypress command but it silently fails. When the test a passes the command goes through just fine. 

```
afterEach(function () {
  if (this.currentTest.state !== 'passed') {
    var testName = this.currentTest.parent.title

    failed.push({testName: testName, message:`${this.currentTest.err.name}:
    ${this.currentTest.err.message}`})
  }
  console.log(failed)
  writeToFixture(failed, 'failed.json')
})
```

```
function writeToFixture (data, fileName) {
  debugger
  cy
    .exec("echo '" + JSON.stringify(data, null, 2) + "'>cypress/fixtures/" + fileName)
}
```

**Comment by brian-mann** [comment]
We are close to supporting custom reporters to basically facilitate what you're trying to do.

@chrisbreiding is working on it right now.

**Comment by chrisbreiding** [comment]
We'll soon have [support for custom reporters](https://github.com/cypress-io/cypress/issues/231), which will allow building data about test failures, but  won't directly solve the issue here. It also will only work in headless mode, which may not be ideal.

The real issue here is that commands stop running after a failure, even if they are in an `after` or `afterEach` hook. This runs contrary to how mocha works, since it will still run those hooks and the code in them after a failure. It's reasonable that a user may need to do cleanup work or reporting (like above) using commands after a failure. We'll need to look into allowing certain or all commands to run in `after` and `afterEach` hooks even after a failure.

**Comment by brian-mann** [comment]
This issue is finally fixed in the next release. Currently when a Cypress test failed, the `after` and `afterEach` hooks would execute but Cypress commands were not being queued properly (and thus would never run).

**Comment by brian-mann** [comment]
Fixed in [`0.17.5`](https://github.com/cypress-io/cypress/wiki/Changelog#0175-10022016).
